src/OfflineSearch.xml

```diff
@@ -0,0 +1,47 @@
+<?xml version="1.0" encoding="utf-8"?>
+<widget id="com.mendix.widget.custom.OfflineSearch.OfflineSearch" offlineCapable="true" needsEntityContext="false" xmlns="http://www.mendix.com/widget/1.0/">
+    <name>Offline Search</name>
+    <description>Search Mendix List views, Template Grids and Data Grids using a single-field search</description>
+    <icon />
+
+    <properties>
+        <property key="targetGridName" type="string" required="true" defaultValue="">
+            <caption>Grid name</caption>
+            <category>Data source</category>
+            <description>The name of the grid on this page</description>
+        </property>
+        <property key="searchMethod" type="enumeration" defaultValue="contains">
+            <caption>Search method</caption>
+            <category>Data source</category>
+            <description>Search method to use</description>
+            <enumerationValues>
+                <enumerationValue key="equals">Equals</enumerationValue>
+                <enumerationValue key="lessThan">Less than</enumerationValue>
+                <enumerationValue key="lessThanOrEquals">Less than or equals</enumerationValue>
+                <enumerationValue key="greaterThan">Greater than</enumerationValue>
+                <enumerationValue key="greaterThanOrEquals">Greater than or equals</enumerationValue>
+                <enumerationValue key="contains">Contains</enumerationValue>
+            </enumerationValues>
+        </property>
+        <property key="searchEntity" type="string" required="false">
```

Comment:
Based on the sample i was sent and based on the looking at the internal API's. This checking is not needed since we only pass the attribute to the `constraints` string or object. 
Still based on the sample, the `entity` in this case is the associated entity to the listview entity. I have looked at the API on how this validation can be done for the associated entity. It is not possible.